src/Bus/MongoBatchRepository.php

```diff
@@ -0,0 +1,270 @@
+            ],
+        );
+
+        return $batch ? $this->toBatch($batch) : null;
+    }
+
+    #[Override]
+    public function store(PendingBatch $batch): Batch
+    {
+        $result = $this->collection->insertOne([
+            'name' => $batch->name,
+            'total_jobs' => 0,
+            'pending_jobs' => 0,
+            'failed_jobs' => 0,
+            'failed_job_ids' => [],
+            // Serialization is required for Closures
+            'options' => serialize($batch->options),
+            'created_at' => new UTCDateTime(Carbon::now()),
+            'cancelled_at' => null,
+            'finished_at' => null,
+        ]);
+
+        return $this->find($result->getInsertedId());
+    }
+
+    #[Override]
+    public function incrementTotalJobs(string $batchId, int $amount): void
+    {
+        $batchId = new ObjectId($batchId);
+        $this->collection->updateOne(
+            ['_id' => $batchId],
+            [
+                '$inc' => [
+                    'total_jobs' => $amount,
+                    'pending_jobs' => $amount,
+                ],
+                '$set' => [
+                    'finished_at' => null,
+                ],
+            ],
+        );
+    }
+
+    #[Override]
+    public function decrementPendingJobs(string $batchId, string $jobId): UpdatedBatchJobCounts
+    {
+        $batchId = new ObjectId($batchId);
+        $values = $this->collection->findOneAndUpdate(
+            ['_id' => $batchId],
+            [
+                '$inc' => ['pending_jobs' => -1],
+                '$pull' => ['failed_job_ids' => $jobId],
+            ],
+            [
+                'projection' => ['pending_jobs' => 1, 'failed_jobs' => 1],
+                'returnDocument' => FindOneAndUpdate::RETURN_DOCUMENT_AFTER,
+            ],
+        );
+
+        return new UpdatedBatchJobCounts(
+            $values['pending_jobs'],
+            $values['failed_jobs'],
+        );
+    }
+
+    #[Override]
+    public function incrementFailedJobs(string $batchId, string $jobId): UpdatedBatchJobCounts
+    {
+        $batchId = new ObjectId($batchId);
+        $values = $this->collection->findOneAndUpdate(
+            ['_id' => $batchId],
+            [
+                '$inc' => ['failed_jobs' => 1],
+                '$push' => ['failed_job_ids' => $jobId],
+            ],
+            [
+                'projection' => ['pending_jobs' => 1, 'failed_jobs' => 1],
+                'returnDocument' => FindOneAndUpdate::RETURN_DOCUMENT_AFTER,
+            ],
+        );
+
+        return new UpdatedBatchJobCounts(
+            $values['pending_jobs'],
+            $values['failed_jobs'],
+        );
+    }
+
+    #[Override]
+    public function markAsFinished(string $batchId): void
+    {
+        $batchId = new ObjectId($batchId);
+        $this->collection->updateOne(
+            ['_id' => $batchId],
+            ['$set' => ['finished_at' => new UTCDateTime(Carbon::now())]],
```

Comment:
Is `Carbon::now()` mainly used to support mocking of time in the test suite? If so, is this worthy of a comment for future readers?